thoth/storages/graph/postgres.py

```diff
@@ -1880,6 +1880,78 @@ def get_analyzed_error_python_package_versions_count_all(
 
     # Unanalyzed Python Packages
 
+    def _construct_si_unanalyzed_python_package_versions_query(
+        self, session: Session, os_name: Optional[str] = None, os_version: Optional[str] = None, python_version: Optional[str] = None, index_url: Optional[str] = None
+    ) -> Query:
+        """Construct query for packages analyzed by solver, but unanalyzed by SI."""
+        index_url = GraphDatabase.normalize_python_index_url(index_url)
+        query = session.query(PythonPackageVersion).filter(
+            PythonPackageVersion.package_version.isnot(None),
+            PythonPackageIndex.url.isnot(None),
+            PythonPackageIndex.enabled.is_(True),
+        )
+
+        if index_url is not None:
+            query = query.filter(PythonPackageIndex.url == index_url)
+
+        # We find all rows that are same in PythonPackageVersion and SIAggregated table.
+        conditions = [PythonPackageVersion.entity_id == SIAggregated.id]
+
+        if os_name:
+            conditions.append(PythonPackageVersion.os_name == os_name)
+
+        if os_version:
+            conditions.append(PythonPackageVersion.os_version == os_version)
+
+        if python_version:
+            conditions.append(
+                PythonPackageVersion.python_version == python_version)
+
+        # Finally filter these out.
+        query = query.filter(~exists().where(and_(*conditions)))
+
+        return query
+
+    def get_si_unanalyzed_python_package_versions_all(
+        self,
+        os_name: Optional[str] = None,
+        os_version: Optional[str] = None,
+        python_version: Optional[str] = None,
+        start_offset: int = 0,
+        count: Optional[int] = DEFAULT_COUNT,
+        distinct: bool = True,
+        randomize: bool = True,
+    ) -> List[Tuple[str, str]]:
+        """Retrieve solved Python package versions in Thoth Database, that are not anaylyzed by SI. 
+        Examples:
+        >>> from thoth.storages import GraphDatabase
+        >>> graph = GraphDatabase()
+        >>> graph.get_si_unanalyzed_python_package_versions_all()
+        [('crossbar', '0.10.0', 'https://pypi.org/simple'), ('tensorflow', '1.11.0', 'https://pypi.org/simple')]
+        """
+        os_version = OpenShift.normalize_os_version(os_name, os_version)
```

Comment:
this line should go in `_construct_si_unanalyzed_python_package_versions_query`

Reply:
Sure!